crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -816,6 +816,10 @@ fn check_rrsig_validity(
     dnskey: RecordRef<'_, DNSKEY>,
     current_time: u32,
 ) -> RrsigValidity {
+    let current_time = SerialNumber(current_time);
+    let expiration = SerialNumber(rrsig.data().sig_expiration());
```

Comment:
Should `sig_expiration()` and `sig_inception()` return a `SerialNumber` instead? The current abstraction boundary doesn't seem very helpful to me.

Reply:
I checked [RFC 2535](https://datatracker.ietf.org/doc/html/rfc2535) for details on the fields of the `SIG` record (as `RRSIG` wraps it). The RFC only mentions in [Appendix B, 3.](https://datatracker.ietf.org/doc/html/rfc2535#page-44) that signature expiration & inception "use serial number ring arithmetic".

Committed a change to refactor to use type `SerialNumber` for fields `expiration` & `inception`.